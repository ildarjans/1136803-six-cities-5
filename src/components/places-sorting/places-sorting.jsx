import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {SortType, sortTypeTitles} from "../../const";
import {offersActionCreator} from "../../store/offers/offers-action";
import {withCollapseToggle} from "../../hocs/with-collapse-toggle";
import {selectSortType} from "../../selectors/selectors";
import {processActionCreator} from "../../store/process/process-action";

export const PlacesSortingComponent = ({sortType, onSortTypeChange, collapsed, onCollapseToggle}) => {
  const handlePlacesOptionClick = (evt) => {
    onSortTypeChange(evt.target.dataset.sortType);
    onCollapseToggle();
  };

  return (
    <form className="places__sorting" action="#" method="get">
      <span className="places__sorting-caption">Sort by</span>
      <span
        className="places__sorting-type"
        tabIndex="0"
        onClick={onCollapseToggle}
      >

        {sortTypeTitles[sortType]}

        <svg className="places__sorting-arrow" width="7" height="4">
          <use xlinkHref="#icon-arrow-select"/>
        </svg>
      </span>

      <ul
        className={`places__options places__options--custom ${!collapsed ? `places__options--opened` : ``}`}>

        {Object.values(SortType).map((type, i) => (
          <li
            key={`${i}-${type}`}
            id={`${i}`}
            className={`places__option ${sortType === type ? `places__option--active` : ``}`}
            data-sort-type={type}
            tabIndex="0"
            onClick={handlePlacesOptionClick}
          >
            {sortTypeTitles[type]}
          </li>
        ))}
      </ul>

    </form>

  );
};

PlacesSortingComponent.propTypes = {
  sortType: PropTypes.string.isRequired,
  onSortTypeChange: PropTypes.func.isRequired,
  onCollapseToggle: PropTypes.func.isRequired,
  collapsed: PropTypes.bool.isRequired
};

const mapStateToProps = (state) => ({
  sortType: selectSortType(state),
});

const mapDispatchToProps = (dispatch) => ({
  onSortTypeChange: (sortType) => {
    dispatch(processActionCreator.changeSortType(sortType));
  }
});

export const PlacesSorting = connect(
    mapStateToProps,
    mapDispatchToProps
)(withCollapseToggle(PlacesSortingComponent));
