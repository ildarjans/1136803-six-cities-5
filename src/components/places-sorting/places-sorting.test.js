import React from "react";
import renderer from "react-test-renderer";
import {PlacesSorting} from "./places-sorting";

const popularSortType = `popular`;

const noop = () => {};

describe(`Should PlacesSortingComponent render correctly`, () => {
  it(`then collapsed`, () => {
    const tree = renderer.create(
        <PlacesSorting
          sortType={popularSortType}
          collapsed={true}
          onSortTypeChange={noop}
          onCollapseToggle={noop}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`then expanded`, () => {
    const tree = renderer.create(
        <PlacesSorting
          sortType={popularSortType}
          collapsed={false}
          onSortTypeChange={noop}
          onCollapseToggle={noop}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
