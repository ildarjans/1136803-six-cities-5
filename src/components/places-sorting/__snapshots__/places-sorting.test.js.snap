// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PlacesSortingComponent render correctly then collapsed 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom "
  >
    <li
      className="places__option places__option--active"
      data-sort-type="popular"
      id="0"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option "
      data-sort-type="asc"
      id="1"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      data-sort-type="desc"
      id="2"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      data-sort-type="top"
      id="3"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;

exports[`Should PlacesSortingComponent render correctly then expanded 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--opened"
  >
    <li
      className="places__option places__option--active"
      data-sort-type="popular"
      id="0"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option "
      data-sort-type="asc"
      id="1"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      data-sort-type="desc"
      id="2"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      data-sort-type="top"
      id="3"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
